src/CppUTest/TestRegistry.cpp

```diff
@@ -50,6 +50,7 @@ void TestRegistry::runAllTests(TestResult& result)
     result.testsStarted();
     for (UtestShell *test = tests_; test != NULL; test = test->getNext()) {
```

Comment:
Can you make setRunIgnored virtual. Let the UtestShell implement return true and move the member variable in the subclass.

You do not need the member variable in the base class and removing that makes it simpler.